File: core/src/main/java/com/emarsys/core/notification/NotificationManagerHelper.java
package com.emarsys.core.notification;

import android.os.Build;

import androidx.annotation.RequiresApi;

import com.emarsys.core.api.notification.ChannelSettings;
import com.emarsys.core.api.notification.NotificationSettings;
import com.emarsys.core.util.Assert;

import java.util.List;
import java.util.Objects;

public class NotificationManagerHelper implements NotificationSettings {

    private NotificationManagerProxy notificationManagerProxy;

    public NotificationManagerHelper(NotificationManagerProxy notificationManagerProxy) {
        Assert.notNull(notificationManagerProxy, "NotificationManagerProxy must not be null!");

        this.notificationManagerProxy = notificationManagerProxy;
    }

    @Override
    public boolean areNotificationsEnabled() {
        return notificationManagerProxy.areNotificationsEnabled();
    }

    @Override
    public int getImportance() {
        return notificationManagerProxy.getImportance();
    }

    @Override
    @RequiresApi(api = Build.VERSION_CODES.O)
    public List<ChannelSettings> getChannelSettings() {
        return notificationManagerProxy.getNotificationChannels();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        NotificationManagerHelper that = (NotificationManagerHelper) o;
        return Objects.equals(notificationManagerProxy, that.notificationManagerProxy);
    }

    @Override
    public int hashCode() {
        return Objects.hash(notificationManagerProxy);
    }
}

File: core/src/main/java/com/emarsys/core/notification/NotificationManagerHelper.kt
package com.emarsys.core.notification

import android.os.Build
import androidx.annotation.RequiresApi
import com.emarsys.core.Mockable
import com.emarsys.core.api.notification.ChannelSettings
import com.emarsys.core.api.notification.NotificationSettings
import java.util.*

@Mockable
class NotificationManagerHelper(private val notificationManagerProxy: NotificationManagerProxy) :
    NotificationSettings {

    override val areNotificationsEnabled: Boolean
        get() = notificationManagerProxy.areNotificationsEnabled

    override val importance: Int
        get() = notificationManagerProxy.importance

    @get:RequiresApi(api = Build.VERSION_CODES.O)
    override val channelSettings: List<ChannelSettings>
        get() = notificationManagerProxy.notificationChannels

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as NotificationManagerHelper
        return notificationManagerProxy == that.notificationManagerProxy
    }

    override fun hashCode(): Int {
        return Objects.hash(notificationManagerProxy)
    }
}